### PR TITLE
Jon strabala patch 3

### PR DESCRIPTION
**Pull request by jon-strabala**
Amarantha - this is a change e.g. I have a new page (to be deleted but I didn't delete it) that MUST replace an old page - this is only 6.5 (and above).  Slack me if it's confusing

**Old: (to delete - I didn't do this)**	
Adding a Couchbase Function
	eventing-adding-function.adoc

**New (to replace):**		
The Eventing Lifecycle
	eventing-lifecycle.adoc

Update all TAGS from "Adding a Couchbase Function" to "The Eventing Lifecycle"
Update all links from "eventing-adding-function.adoc" to "eventing-lifecycle.adoc"

Needs  image “lifecycle-overview.png”

### CODE REVIEW

**Comment by simon-dew** [comment]
I have removed the `eventing-adding-function` page and replaced all references to this page with references to the new `eventing-lifecycle` page. I have also added a page redirect from `eventing-adding-function` to `eventing-lifecycle`. Finally, I have moved all new images to the correct location for Antora to pick them up. 

Note that GitHub detects that `eventing-lifecycle` is a partial rewrite of `eventing-adding-function` and depicts it as such in the [Files changed](https://github.com/couchbase/docs-server/pull/1152/files) page. This is why some of the suggested changes are marked as outdated — they're not. Hopefully you'll be able to accept all the changes as necessary.

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 1]
```diff
@@ -1,8 +1,15 @@
-= Adding a Couchbase Function
+= The Eventing Lifecyle
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
= The Eventing Lifecycle
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
 
 [abstract]
-To add a new Function, proceed as follows:
+To add a new Eventing Function and to breifly explore the Eventing Lifecycle
+
+[#eventing_lifecyle_image]
+image::lifecycle_overview_notitle.png[,%100]
+
+This is a quick introduction to Eventing via creating a simple Eventing Handler and subsequently excercising the Eventing 
+handler's functionality and operations across its typical lifecyle (for a deeper insight refer to the more detailed xref:eventing-examples.adoc[Eventing Examples])
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
This is a quick introduction to Eventing via creating a simple Eventing Handler and subsequently exercising the Eventing 
Handler's functionality and operations across its typical lifecycle. For a deeper insight refer to the more detailed xref:eventing-examples.adoc[Eventing Examples].
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+[#eventing_lifecyle_image]
+image::lifecycle_overview_notitle.png[,%100]
+
+This is a quick introduction to Eventing via creating a simple Eventing Handler and subsequently excercising the Eventing 
+handler's functionality operations across its typical lifecyle (for a deeper insight refer to the more detailed xref:eventing-examples.adoc[Eventing Examples])
+
+[#eventing_examples_preparations]
+. From the Couchbase Web Console > *Eventing* page, click *ADD FUNCTION*.
+. An empty *ADD FUNCTION* dialog with defaults is shown below:
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
. From the Couchbase Web Console > *Eventing* page, click *ADD FUNCTION*.
+
An empty *ADD FUNCTION* dialog is shown:
```
The second point is not really a step by itself

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
++
+image::addfunc_01_empty_settings.png[,484]
+. In the *ADD FUNCTION* dialog, for individual Function elements allows the developer to provide the below information:
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
+
The *ADD FUNCTION* dialog enables the developer to provide the following information:
```
Not really a step by itself

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+|===
+
+. *Prerequisites*: In order to show the basic life cycle of adding an Eventing Function we need a 'source' bucket and a metadata *metadata*. You must have a bucket typically called *metadata* used solely by Eventing if it is misisng create it with a minimum size of 100MB.  For the source bucket we will use 'travel-sample' sample document set. 
++
+If you don’t already have the sample bucket *beer-sample* listed in the *Couchbase Web Console* > *Buckets* page you can load this document set as follows:
+
+** Access the *Couchbase Web Console* > *Settings* page
+** Select the *Sample Buckets* in the upper right banner. 
+** Check *beer-sample* checkbox.
+** Click *Load Sample Data*.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]: The prerequisites should come _before_ the procedure, i.e. before `#eventing_examples_preparations`. If the users have followed the steps to this point, and then find that they don't have a `metadata` bucket or a `beer-sample` bucket, they'll have to cancel the dialog and start again, which is a bad user experience.

Also please note the following suggestions:
```suggestion
*Prerequisites*: In order to show the basic life cycle of adding an Eventing Function we need a source bucket and a metadata bucket. You must have a metadata bucket, typically called *metadata*, to be used solely by Eventing -- if it is missing, create it with a minimum size of 100MB. For the source bucket we will use the *beer-sample* sample document set. 
+
If you don’t already have the sample bucket *beer-sample* listed in the *Couchbase Web Console* > *Buckets* page you can load this document set as follows:

** Access the *Couchbase Web Console* > *Settings* page
** Select the *Sample Buckets* in the upper right banner. 
** Check *beer-sample* checkbox.
** Click *Load Sample Data*.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+. To get to the next screen *ADD FUNCTION* dialog, as follows:
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
. In the *ADD FUNCTION* dialog, enter the following information:
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
++
+image::addfunc_02_settings.png[,484]
+. After providing all the required information in the *ADD FUNCTION* dialog, click *Next: Add Code*.
+The *delete_orphaned_txns* dialog appears.
+** The *my_evt_function* dialog initially contains a placeholder code block.
+You will accept the default for your *my_evt_function code*.
++
+image::addfunc_03_editor_with_default.png[,100%]
+** Click *Save*.
+** To return to the Eventing screen, click the '*< back to Eventing*' link (below the editor) or click *Eventing* tab.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
** To return to the Eventing screen, click the '*< back to Eventing*' link (below the editor) or click the *Eventing* tab.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+. Click on the Function name.
++
+image::addfunc_04_newundeployed.png[,100%]
+Additional controls are now displayed, the controls are:
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
Additional controls are now displayed. The controls are:
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+ ** *Delete*: Deletes the Eventing Function from the system.
+ ** *Export*: Exports the EventingFunction as a JSON document.
+ ** *Deploy*: Deploys the EventingFunction, making it active across the cluster.
+ ** *Pause*: Pauses the EventingFunction, making it active across the cluster (only allowed if Deployed).
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
 ** *Export*: Exports the Eventing Function as a JSON document.
 ** *Deploy*: Deploys the Eventing Function, making it active across the cluster.
 ** *Pause*: Pauses the Eventing Function, making it active across the cluster (only allowed if Deployed).
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+ ** *Edit JavaScript*: Allows edits to be made on the Eventing Function, in an edit dialog (only allowed when Paused or Undeployed).
+
+. From the *Eventing* screen, click *Deploy*.
+** In the *Confirm Deploy Function* dialog, select *Everything from the Feed boundary* option.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
** In the *Confirm Deploy Function* dialog, select *Everything* from the *Feed boundary* option.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
++
+The Feed Boundary determines whether documents previously in existence needs to be included in the Function's activities: the options are *Everything* and *From now*. The *Everything* option invokes a Function on all mutations available in the cluster The *From now* option invokes a Function during future instances of data mutation, post Function deployment.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
The Feed Boundary determines whether documents previously in existence need to be included in the Function's activities: the options are *Everything* and *From now*. The *Everything* option invokes a Function on all mutations available in the cluster. The *From now* option invokes a Function during future instances of data mutation, post Function deployment.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+** Click *Deploy Function*.
+
+. The Eventing function is deployed and starts running within a few seconds. From this point, the defined Function is executed on all existing documents and on subsequent mutations.
++
+This simple example is listing to the bucket 'beer-sample' which has 7,303 documents.  The only thing the function does is print the ID of each document.
+
+. Once the Eventing Function is fully deployed it's status will change form *deploying...* to a status of *deployed*
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
. Once the Eventing Function is fully deployed its status will change from *deploying...* to a status of *deployed*. 
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+The Feed Boundary determines whether documents previously in existence needs to be included in the Function's activities: the options are *Everything* and *From now*. The *Everything* option invokes a Function on all mutations available in the cluster The *From now* option invokes a Function during future instances of data mutation, post Function deployment.
+** Click *Deploy Function*.
+
+. The Eventing function is deployed and starts running within a few seconds. From this point, the defined Function is executed on all existing documents and on subsequent mutations.
++
+This simple example is listing to the bucket 'beer-sample' which has 7,303 documents.  The only thing the function does is print the ID of each document.
+
+. Once the Eventing Function is fully deployed it's status will change form *deploying...* to a status of *deployed* 
+
+. Verify actually worked by clicking the *Log* link that appeared after you deployed the Eventing Function in the right hand side of the screen.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
. Verify the deployment by clicking the *Log* link that appeared after you deployed the Eventing Function in the right hand side of the screen.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+** A dialog showing the *Function Log - my_evt_function* will appear with the most recent logging information (in reverse order with the most recent lines first).
++
+image::addfunc_05_logs_emitted.png[,700]
+** Click *Close*
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
** Click *Close*.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+
+. To pause and resume a function (you can then edit and update the function without missing a mutation)  
+** Click *Pause*
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
** Click *Pause*.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+image::addfunc_05_logs_emitted.png[,700]
+** Click *Close*
+
+. To pause and resume a function (you can then edit and update the function without missing a mutation)  
+** Click *Pause* 
+** In the *Confirm Pause Function* dialog
+*** Click *Pause Function*.
+** The Eventing function will now pause.
+** Wait for the "paused" state.
+** Click *Resume*
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
** Click *Resume*.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+*** Click *Pause Function*.
+** The Eventing function will now pause.
+** Wait for the "paused" state.
+** Click *Resume* 
+** In the *Confirm Resume Function* dialog
+*** Click *Resume Function*.
+** The Eventing function will now resume.
+
+. To remove the Eventing Function *my_evt_function*  
+** Click *Undeploy*
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
** Click *Undeploy*.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+*** Click *Resume Function*.
+** The Eventing function will now resume.
+
+. To remove the Eventing Function *my_evt_function*  
+** Click *Undeploy* 
+** In the *Confirm Undeploy Function* dialog
+*** Click *Undeploy Function*.
+** The Eventing function will now undeploy.
+** Wait for the "undeployed" state.
+** Click *Delete*
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
** Click *Delete*.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 19]
```diff
@@ -98,7 +105,9 @@ The Feed Boundary determines whether documents previously in existence needs to
 
 . The Eventing function is deployed and starts running within a few seconds. From this point, the defined Function is executed on all existing documents and on subsequent mutations.
 +
-This simple example is listing to the bucket 'beer-sample' which has 7,303 documents.  The only thing the function does is print the ID of each document.
+image::input-output-overview-6.5.png[,%100]
++
+This simple example here only has a log(....) statement in the *OnUpdate* handler that will merely list items in the bucket 'beer-sample', e.g. 7,303 documents. Basically only thing the function does is print the key or ID of each document.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
This simple example only has a log(....) statement in the *OnUpdate* handler that will merely list items in the bucket 'beer-sample', i.e. 7,303 documents. Basically the only thing the function does is print the key or ID of each document.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 20]
```diff
@@ -0,0 +1,138 @@
+= The Eventing Lifecyle
+:page-edition: Enterprise Edition
+
+[abstract]
+To add a new Eventing Function and to breifly explore the Eventing Lifecycle
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]:
```suggestion
This page shows how to add a new Eventing Function and briefly explores the Eventing Lifecycle.
```

File: modules/eventing/pages/eventing-lifecycle.adoc [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
+ ** For the *Bindings* option, don't add any bindings (we will merely be logging messages).
++
+image::addfunc_02_settings.png[,484]
+. After providing all the required information in the *ADD FUNCTION* dialog, click *Next: Add Code*.
+The *delete_orphaned_txns* dialog appears.
+** The *my_evt_function* dialog initially contains a placeholder code block.
+You will accept the default for your *my_evt_function code*.
++
+image::addfunc_03_editor_with_default.png[,100%]
+** Click *Save*.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-lifecycle.adoc]: I found that the **Save** button was unavailable. I had to make at least one change and undo it before I could save the boilerplate code. Please check if this step is necessary and remove it if not.

**Review by simon-dew — CHANGES_REQUESTED** [review]
Again, sorry for the large number of suggestions. A lot of them relate to text from the previous `eventing-adding-function` page.

**Comment by simon-dew** [comment]
Okay, that didn't work, you can't accept suggestions to an outdated version of the file. I'll have to push another commit containing my suggested changes for the legacy `eventing-adding-function` text.

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 3 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 4 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 5 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 6 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 7 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 8 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 9 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 10 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 11 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 13 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 15 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 16 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 17 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 18 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 14 above)
Comment (simon-dew) [reply modules/eventing/pages/eventing-lifecycle.adoc]: Added to separate commit

**Comment by simon-dew** [comment]
Sorry for the delay @jon-strabala. If you could please look at the unresolved comments. There are only about 5 of them now.

File: modules/eventing/pages/eventing-lifecycle.adoc (hunk 21 above)
Comment (jon-strabala) [reply modules/eventing/pages/eventing-lifecycle.adoc]: You are correct it comes in as a default and a Save is not neeeded.

**Review by simon-dew — APPROVED** [review]
LGTM 👍

**Pull request merged**